src/api/WorkspaceService.ts

```diff
@@ -0,0 +1,41 @@
+import { ServicesMap, ServicesRequest } from './methods/services';
+import { RegisterServiceRequest } from './methods/registerService';
+import { ComponentsMap, ComponentsRequest } from './methods/components';
+
+/**
+ * Workspace is the core service of Capsula Hub, it is responsible for :
+ * - Allowing services to register themselves (REGISTER)
+ * - Letting services and components communicate together (SERVICES/COMPONENTS)
+ * - Getting their own configuration (CONFIG)
+ */
+export interface WorkspaceService {
+  /**
+   * Getting a map of promises to each service that has been loaded in the workspace
+   * Reject in case :
+   * - Invalid request
+   * - Workspace not started yet
```

Comment:
How can Worspace not be started, if we return it in a resolved promise of **createWorkspace** only when it has already started. Or have I misunderstood the logic?

Reply:
I removed all this rejections